src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+            }
+
+            return $this->chatClient->postMessage($command->getRoom(), "Dunno what happened but I couldn't set the reminder.");
+        });
+    }
+
+    private function getAllReminders(Command $command): Promise
+    {
+        return resolve(function() use($command) {
+            $message = "Registered reminders are:";
+
+            $reminders = yield $this->storage->getAll($command->getRoom());
+            if($reminders === []){
+                return $this->chatClient->postMessage($command->getRoom(), "There aren't any scheduled reminders.");
+            }
+
+            $timeouts = [];
+            foreach ($reminders as $item => $value) {
+                $text = $value['text'];
+                $user = $value['username'];
+                $key  = base_convert($value['id'], 10, 36);
+                $timestamp = $value['timestamp'];
+
+                if (!$timestamp) {
+                    return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!");
+                }
+                $seconds = $timestamp - time();
+
+                if ($seconds <= 0) {
+                    $timeouts[] = $key;
+                    continue;
+                }
+
+                $message .= sprintf(
+                    "\n%s %s %s %s %s %s - %s - %s ",
+                    Chars::BULLET,
+                    $text,
+                    Chars::RIGHTWARDS_ARROW,
+                    "Id: " . $key,
+                    Chars::RIGHTWARDS_ARROW,
+                    date('l, dS F Y H:i (e)', $timestamp),
+                    'Set by ' . $user,
+                    'Seconds left: ' . $seconds
+                );
+            }
+
+            if(count($timeouts) !== count($reminders)){
+                return $this->chatClient->postMessage($command->getRoom(), $message);
+            }
+        });
+    }
+
+    /**
+     * Handle a command message
+     *
+     * According to http://www.strawpoll.me/11212318/r this plugin will support the following commands
+     * !!<reminder|remind> remind this <in|at> <time>
+     * !!at <time> remind this
+     * !!in <time> remind this
+     *
+     * @param Command $command
+     * @return Promise
+     */
+    public function handleCommand(Command $command): Promise
+    {
+        return resolve(function() use($command) {
+
+            if ($command->hasParameters() === false) {
+                return $this->chatClient->postMessage($command->getRoom(), self::USAGE);
+            }
+
+            /* $command->getParameter(0) can be: list | examples | flush | unset | <text> | <time> */
+            $textOrCommand = $command->getParameter(0);
+            $commandName = $command->getCommandName(); // <reminder|in|at>
+            switch ($commandName){
+                case 'reminder': case 'remind': break;
+                case 'in': case 'at': return yield $this->setReminder($command); break;
```

Comment:
Don't put it all on a single line:

    case 'in':
    case 'at':
        return yield $this->setReminder($command);